recommendations/local-resource-conservation.md

```diff
@@ -0,0 +1,333 @@
+MAY choose to limit its exposure to HTLCs that put it at risk of a denial of 
+service attack.
+
+We define the following parameters:
+* `protected_slot_count`: defines the number of HTLC slots that are reserved 
+  for endorsed HTLCs from peers with sufficient reputation (default: 0.5 * 
+  remote peer's `max_accepted_htlcs`).
+* `protected_liquidity_portion`: defines the portion of liquidity that is 
+  reserved for endorsed HTLCs from peers with sufficient reputation (default: 
+  0.5).
+
+A node implementing resource bucketing limits exposure on its outgoing channel:
+* MUST choose `protected_slot_count` <= the remote channel peer's 
+  `max_accepted_htlcs`.
+* MUST choose `protected_liquidity_portion` in [0;1].
+
+For each `update_add_htlc` proposed by an incoming link:
+* If `endorsed` is non-zero AND the incoming link has sufficient local 
+  reputation for the HTLC (see [Recommendations for Reputation Scoring](#recommendations-for-reputation-scoring)):
+  * SHOULD forward the HTLC as usual.
+  * SHOULD set `endorsed` to 1 in the outgoing `update_add_htlc`.
+* Otherwise: 
+  * SHOULD reduce the remote peer's `max_accepted_htlcs` by 
+    `protected_slot_count` for the purposes of the proposed HTLC.
+  * SHOULD reduce the `max_htlc_value_in_flight` by 
+    `protected_liquidity_portion` * `funding_satoshis`.
+  * SHOULD set `endorsed` to `0` in the outgoing `update_add_htlc`.
+
+## Implementation Notes
+
+### Decaying Average
+Rolling windows specified in this write up may be implemented as a decaying 
+average to minimize the amount of data that needs to be stored per-link. In 
+flight HLTCs can be accounted for separately to this calculation, as the node 
+will already have data for these HTLCs available.
+
+Track the following values for each rolling window: 
+* `last_update`: stores the timestamp of the last update to the decaying 
+    average, expressed in seconds.
+* `decaying_average`: stores the value of the decaying average.
+* `decay_rate`: a constant rate of decay based on the rolling window chosen, 
+  calculated as: `((1/2)^(1/window_length_seconds/2))`.
+
+To update the `decaying_average` at time `t`:
+* `decaying_average` = `decaying_average` * `decay_rate` ^ `last_update_diff`.
+* `last_update` = `t`.
+
+When assessing whether a link has sufficient reputation for a HTLC: 
+* MUST update `decaying_average` as described above.
+
+When updating the `decaying_average` to add the `effective_fee` of a newly 
+resolved HTLC at timestamp `t`: 
+* MUST update `decaying_average` as described above.
+* `decaying_average` = `decaying_average` + `effective_fee`
+
+### Multiple Channels
+If the local node has multiple channels open with the incoming and outgoing 
+links involved in a HTLC forward, they can be grouped as follows: 
+* Incoming link: consider `incoming_link_revenue` across all channels with the 
+  peer, as the sum of all forwarded HTLCs represents the total contribution. 
```

Comment:
With more channels, that can create more damage in a slot based attack. Not sure it this is a huge problem.

Reply:
MAY but must look at all in flight across all channels if you do that.